### DISCUSSION TITLE
Create and publish documentation site to Github Pages?

### DISCUSSION THREAD

**Discussion by marcospereira** [casid/jte#286] [Ideas]
Hey folks,

The documentation right now is somewhat split into two different files:

1. `README.md`
2. `DOCUMENTATION.md`

With some duplications between them, documentation being a single file covering different topics, and navigation being directly to GitHub web page. Also, while jte supports both Java and Kotlin, the docs don't cover Kotlin (or offer kte examples).

To be honest, none of these are big problems, which is a statement of how simple and well-designed jte is. But I thought we could improve the docs experience by using a docs-site generator and publishing it to GitHub pages.

**Comment by marcospereira** [comment]
Okay, here is a try using [Material for MkDocs](https://squidfunk.github.io/mkdocs-material/).

https://marcospereira.github.io/jte/

There are many details to fix/improve, but it is a good start. A few nice things there already:

1. Light/dark mode
2. Code tabs
    <img width="303" alt="Screenshot 2023-10-10 at 6 11 53 PM" src="https://github.com/casid/jte/assets/4576/460c62c5-1f0f-447c-95ae-f7f9f483979b">
3. [Note/info/tip/warning boxes](https://squidfunk.github.io/mkdocs-material/reference/admonitions/?h=ad#supported-types):
    <img width="379" alt="Screenshot 2023-10-10 at 6 13 19 PM" src="https://github.com/casid/jte/assets/4576/16c91bff-7104-4c2f-8352-a7f19c583c47">
4. Search
    <img width="374" alt="Screenshot 2023-10-10 at 6 15 01 PM" src="https://github.com/casid/jte/assets/4576/44aba263-fe0e-4025-a60b-4dcec797be95">
5. All seamless published by a Github Action: https://github.com/marcospereira/jte/blob/docs/poc-mkdocs/.github/workflows/publish-docs.yml.

@casid @edward3h, WDYT? Is it worth to invest extra time to get it done?

**Reply by casid** [reply]
@marcospereira that's fantastic!

There currently is a separate Github project to generate the jte website https://github.com/casid/jte-website/, which populates https://jte.gg/

It would be really cool to have the documentation you generated available on jte.gg. I would say we could replace the current landing page entirely with the documentation you've created. This would also spare me from bumping the version on the jte website each time we're doing a new release :-)

If we add your Github Action to the jte repository, I'd think I then would just need to tell Github to publish the jte website from the jte repository and no longer from jte-website.

**Reply by casid** [reply]
I had a closer look at your repository. Is there a way that we can avoid the duplication of README.md / DOCUMENTATION.md and the markdown in docs?

**Reply by marcospereira** [reply]
> It would be really cool to have the documentation you generated available on jte.gg. I would say we could replace the current landing page entirely with the documentation you've created. This would also spare me from bumping the version on the jte website each time we're doing a new release :-)

Yeah, that sounds like a good plan. :-)

> If we add your Github Action to the jte repository, I'd think I then would just need to tell Github to publish the jte website from the jte repository and no longer from jte-website.

I don't know for sure how the process works, but it seems that you need to change the domain configuration for GH Pages from one repo to another. 

> I had a closer look at your repository. Is there a way that we can avoid the duplication of README.md / DOCUMENTATION.md and the markdown in docs?

I would keep `DOCUMENTATION.md` file with only a note pointing to the new docs location. Search engines index that file, so having it there as a pointer would be good. I would also keep `README.md` minimal and point to the docs. I can give it a try later.

**Comment by marcospereira** [comment]
@casid, I've just created https://github.com/casid/jte/pull/288. Closing the discussion since we can make decisions there. :-)

**Reply by casid** [reply]
Thanks, will do!
